### PR TITLE
Allow repository math Filter and QueryModifier.

### PR DESCRIPTION
**Pull request by Strontium-90**
Related to #116 
Also allow just apply specification without executing query. And apply it to different QueryBuilders with different dqlAliases.

### CODE REVIEW

File: src/EntitySpecificationRepositoryTrait.php [hunk 1]
```diff
@@ -124,4 +121,33 @@ public function getAlias()
...
         return $this->alias;
     }
+
+    /**
+     * @param QueryBuilder         $queryBuilder
+     * @param Filter|QueryModifier $specification
+     *
+     * @throws \InvalidArgumentException
+     */
+    protected function applySpecification(QueryBuilder $queryBuilder, $specification = null, $alias = null)
```
Comment (Nyholm) [inline src/EntitySpecificationRepositoryTrait.php]: Why would you like to have $alias as third parameter?

**Comment by Nyholm** [comment]
Thank you for this PR. It all looks good. Just had a minor comment.

File: src/EntitySpecificationRepositoryTrait.php (hunk 1 above)
Comment (Strontium-90) [inline src/EntitySpecificationRepositoryTrait.php]: I can have 2 QueryBuiders in one repository method for main and sub query. And I want apply specification to sub-query.

Comment (Nyholm) [inline src/EntitySpecificationRepositoryTrait.php]: Sure, Just add a `@param` comment about the alias parameter and I'm happy to merge

Comment (Strontium-90) [inline src/EntitySpecificationRepositoryTrait.php]: done

**Comment by Nyholm** [comment]
Thank you!

**Pull request merged**